src/Depfile.cpp

```diff
@@ -199,6 +214,22 @@ tokenize(std::string_view file_content)
         }
       }
       break;
+    case '"':
+      // quotes should take everything until next quotes
+      // skip the first quote
+      ++p;
+      while (p < length) {
+        const char next = file_content[p];
+        if (next == '"') {
+          // skip the last quote
+          ++p;
+          break;
+        } else {
+          token.push_back(next);
+          ++p;
+        }
+      }
+      continue;
```

Comment:
This doesn't match the behavior of GNU make, at least not version 4.3:

```console
% echo '"foo1 foo2": "foo3"' >test.mk
% make -np -f test.mk | grep foo     
make: *** No rule to make target '"foo3"', needed by '"foo1'.  Stop.
.DEFAULT_GOAL := "foo1
"foo3":
foo2": "foo3"
"foo1: "foo3"
```

Note that `"foo1 foo2"` is parsed as two different tokens (`"foo1` and `foo2"`), and those tokens are interpreted verbatim when evaluating the makefile.

Reply:
All my problems are coming from the tasking support rebase and enhancement (to support the depend mode which provides an extra 30% improvment).

The tasking compiler generates the dependency file with the following changes compared to a standard gcc compiler:
- extra space between target and colon -> supported by gnu make (on the versions I tested 3.82 and 4.3)
- no space between colon and pre-requisite -> supported by gnu make (on the versions I tested 3.82 and 4.3)
- quotes around the file names -> you highlighted that this is not supported by make so I tried to understand why my build environment which is make-based still works, and I discovered that there is a post processing of the generated dependency files (bouhh!! ugly).  

I tried to enhance as much as possible the core of the ccache tool when doing my tests and I thought that all those changes were meaningful for a broader support even for the quotes because they are a standard way of specifying names without escaping characters.

Of course, this is your tool, you can decide to accept this or not.  Could you just tell me what you decide (make the quoting part of the tasking support changes or part of the core) and I will comply to it?

thanks

Reply:
> The tasking compiler generates the dependency file with the following changes compared to a standard gcc compiler:
> 
> * extra space between target and colon -> supported by gnu make (on the versions I tested 3.82 and 4.3)
> * no space between colon and pre-requisite -> supported by gnu make (on the versions I tested 3.82 and 4.3)

That's fine (as mentioned in https://github.com/ccache/ccache/pull/1166#issuecomment-1261313008).

> I thought that all those changes were meaningful for a broader support even for the quotes because they are a standard way of specifying names without escaping characters.

Right, but since that's not how quotes work in makefiles (which a `.d` file is), we can't add such parsing to the common code path since that would introduce bugs for the common GCC/Clang case[^1].

[^1]:  For example, parsing a dependency file generated by  `gcc -MD -c 'exa"mple.c'` will be interpreted incorrectly.

Reply:
Okay, I have removed that change and moved it to the tasking specific behavior

Reply:
oops, sorry, please do not merge that because now we have an issue with the C: which is seen as a : delimiter even though there are quotes (I removed the quotes mechanism)